reflect/reflect.go

```diff
@@ -41,15 +42,62 @@ func StructToMap(val interface{}) (map[string]interface{}, bool) {
 // MapToStruct converts a map to a struct. It is the inverse of the StructToMap
 // function. For details see StructToMap.
 func MapToStruct(m map[string]interface{}, struc interface{}) error {
+	//fmt.Printf("Input map: %+v\n", m)
+	//fmt.Printf("Input struc: %+v\n", struc)
 	val := r.Indirect(r.ValueOf(struc))
 	sinfo := getStructInfo(val)
+	//fmt.Printf("sinfo: %+v\n", sinfo)
 	for k, v := range m {
+		//fmt.Printf("k: %+v  v: %+v\n", k, v)
 		if info, ok := sinfo.FieldsMap[k]; ok {
+			//fmt.Printf("info: %+v\n", info)
 			structField := val.Field(info.Num)
-			if structField.Type().Name() == r.TypeOf(v).Name() {
+			//fmt.Printf("type struct: %q, %q, %q\n", structField.Type(), structField.Type().Name(), structField.Kind())
+			//fmt.Printf("type value: %q\n", r.TypeOf(v).Name())
+			//fmt.Printf("value: %+v\n", r.ValueOf(v))
+			if structField.Kind().String() == "slice" || r.TypeOf(v).Kind().String() == "slice" {
```

Comment:
This is especially hard to follow
- please set appropriate vars for frequently used method chains (for instance r.TypeOf(v) , r.TypeOf(v).Elem() ...
- must be possible to have nicer names than r,k,v,m,struct,val
- please comment subtleties